modules/dns/dns_plugin.rb

```diff
@@ -6,6 +6,13 @@ class Plugin < ::Proxy::Plugin
     uses_provider
     default_settings :use_provider => 'dns_nsupdate', :dns_ttl => 86_400
     plugin :dns, ::Proxy::VERSION
+    capabilities [
+      :TYPE_A,
+      :TYPE_AAAA,
+      :TYPE_CNAME,
+      :TYPE_PTR,
+      :TYPE_SRV,
+    ]
```

Comment:
Nicer DSL could be:

```
capability :DNS_TYPE_A
```

I'd suggest to prefix capabilities with generic module/api name so when there is "TYPE_A" added into DHCP at some point we don't get a name clash.

Lowercase might look better when presented together with list of features, just a thought.

Reply:
Perhaps. I've also considered something like disabling capabilities from a provider. For example the libvirt provider doesn't support some DNS types that we have in core. Currently it just silently drops them which is not nice.